app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -39,27 +41,34 @@ def parent_cluster
     rp = parent_resource_pool
     rp && rp.detect_ancestor(:of_type => "EmsCluster").first
   end
-  alias_method :owning_cluster, :parent_cluster
-  alias_method :ems_cluster, :parent_cluster
+  alias owning_cluster parent_cluster
+  alias ems_cluster parent_cluster
 
   def disconnect_storage(_s = nil)
-    vm_disks = collect_disks
-
-    if vm_disks.blank?
-      storage = nil
-    else
-      vm_storages = ([storage] + storages).compact.uniq
-      storage = vm_storages.select { |store| !vm_disks.include?(store.ems_ref) }
+    unless active?
+      return
```

Comment:
return unless active?
(this is how you did it in the #62 PR and it looks better :))